sentry-core/src/main/java/io/sentry/core/SentryClient.java

```diff
@@ -123,6 +142,31 @@ public SentryId captureEvent(SentryEvent event, @Nullable Scope scope, @Nullable
     return event.getEventId();
   }
 
+  @Override
+  public void captureSession(@NotNull Session session) throws IOException {
+    if (!options.isEnableSessionTracking()) {
+      options.getLogger().log(SentryLevel.WARNING, "Session tracking is disabled in the options.");
+      return;
+    }
```

Comment:
this should be in the `Hub` eh

Reply:
Agreed this makes sense to be in the Hub. Perhaps on both sides since this is public API now (users can call `client.captureSession(new Session)`. Ideally this wouldn't be public though, if we can yank it from the interface would be better